libs/ledger/include/ledger/chain/main_chain.hpp

```diff
@@ -118,7 +120,7 @@ class MainChain
   };
 
   // Construction / Destruction
-  explicit MainChain(Mode mode = Mode::IN_MEMORY_DB);
+  explicit MainChain(std::unique_ptr<BloomFilter> bloom_filter, Mode mode = Mode::IN_MEMORY_DB);
```

Comment:
I think that it would be cleaner for the moment to allow this bloom filter to be optional i.e. it can be a defaulted to a null parameter

Reply:
I'll default it to the DummyBloomFilter, which is the null implementation